apps/src/blockTooltips/DropletAutocompleteParameterTooltipManager.js

```diff
@@ -88,15 +88,21 @@ DropletAutocompleteParameterTooltipManager.prototype.onCursorMovement_ = functio
 DropletAutocompleteParameterTooltipManager.prototype.showParamDropdownIfNeeded_ = function (editor, paramInfo) {
   // Check the dropletConfig to see if we can find dropdown info for this parameter
   var dropdownList;
+  // TODO: Use getAllAvailableDropletBlocks() if we need dropdowns on
+  // builtinConfigBlocks or standardConfig blocks
   this.dropletTooltipManager.dropletConfig.blocks.forEach(function (block) {
-    if (block.func === paramInfo.funcName && block.dropdown) {
+    if (!block.noAutocomplete &&
+        (!this.dropletTooltipManager.blockFilter ||
+          typeof this.dropletTooltipManager.blockFilter[block.func] !== 'undefined') &&
+        block.func === paramInfo.funcName &&
+        block.dropdown) {
```

Comment:
Agreed on simplifying. Will take another pass.

I use `typeof a !== 'undefined'` because `a !== undefined` has issues when a variable was named 'undefined' - and the typeof approach works in all cases without the gotchas: http://stackoverflow.com/questions/4725603/variable-undefined-vs-typeof-variable-undefined